okhttp/src/main/kotlin/okhttp3/internal/-UtilCommon.kt

```diff
@@ -58,10 +58,10 @@ internal val UNICODE_BOMS =
     // UTF-16LE.
     "fffe".decodeHex(),
     // UTF-32BE.
-    "0000ffff".decodeHex(),
+    "0000feff".decodeHex(),
     // UTF-32LE.
-    "ffff0000".decodeHex(),
-  )
+    "fffe0000".decodeHex(),
+  ).sortedByDescending { it.size }
```

Comment:
Please use an explicit order. Arbitrarily sorting breaks the indices used on the matching side, which will also need changed to match the new order.